src/Components/SignUpPage.jsx

```diff
@@ -0,0 +1,147 @@
+import { useState } from "react";
+import "./signUpPage.scss";
+import { account, ID } from "../lib/appwrite";
+import { validateEmailWithRegex } from "../lib/utils/validateEmailWithRegex";
+import { validatePasswordWithRegex } from "../lib/utils/validatePasswordWithRegex";
+
+export default function SignUpPage() {
+  const [showPassword, setShowPassword] = useState(false);
+  const [nameError, setNameError] = useState(true);
+  const [emailError, setEmailError] = useState(true);
+  const [passwordError, setPasswordError] = useState(true);
+
+  // Toggle Show password Button handler //
+  function togglePassword() {
+    setShowPassword(!showPassword);
+  }
+
+  // Create Form SignUp submit button handler to access the all form data in once. //
+  const signUpButtonHandler = async (event) => {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const formDataEntries = formData.entries();
+    const formDataObject = Object.fromEntries(formDataEntries);
+
+    // Email input validation function call. //
+    const isValidEmail = validateEmailWithRegex(formDataObject.email);
+    setEmailError(isValidEmail);
+
+    //  Password input validation function call. //
+    const isValidPassword = validatePasswordWithRegex(formDataObject.password);
+    setPasswordError(isValidPassword);
```

Comment:
this should be setPasswordError(!isValidPassword) bcoz the isValidPassword will return true if password is valid that means the passwordError become true if password is valid which is incorrect. It should be the opposite

Reply:
fixed